var/spack/repos/builtin/packages/py-poetry/package.py

```diff
@@ -12,52 +12,70 @@ class PyPoetry(PythonPackage):
+    depends_on("py-keyring@24", when="@1.6.1", type=("build", "run"))
     depends_on("py-keyring@21.2.0:", when="@1.1.13:", type=("build", "run"))
-    depends_on("py-packaging@20.4:20", when="@:1.1", type=("build", "run"))
+    depends_on("py-keyring@21.2.0:21", when="@1.1.12 ^python@3.6:3", type=("build", "run"))
     depends_on("py-packaging@20.4:", when="@1.2:", type=("build", "run"))
+    depends_on("py-packaging@20.4:20", when="@:1.1", type=("build", "run"))
     depends_on("py-pexpect@4.7:4", type=("build", "run"))
-    depends_on("py-pkginfo@1.4:1", when="@:1.1", type=("build", "run"))
+    depends_on("py-pkginfo@1.9.4:1", when="@1.6.1", type=("build", "run"))
     depends_on("py-pkginfo@1.5:1", when="@1.2:", type=("build", "run"))
-    depends_on("py-platformdirs@2.5.2:2", when="@1.2:", type=("build", "run"))
+    depends_on("py-pkginfo@1.4:1", when="@:1.1", type=("build", "run"))
+    depends_on("py-platformdirs@3", when="@1.6.1", type=("build", "run"))
+    depends_on("py-platformdirs@2.5.2:2", when="@1.2", type=("build", "run"))
+    depends_on("py-requests@2.26:2", when="@1.6.1", type=("build", "run"))
     depends_on("py-requests@2.18:2", type=("build", "run"))
-    depends_on("py-requests-toolbelt@0.9.1:0.9", type=("build", "run"))
-    depends_on("py-shellingham@1.1:1", when="@:1.1", type=("build", "run"))
+    depends_on("py-requests-toolbelt@0.9.1:1", when="@1.6.1", type=("build", "run"))
+    depends_on("py-requests-toolbelt@0.9.1:0.9", when="@:1.2", type=("build", "run"))
     depends_on("py-shellingham@1.5:1", when="@1.2:", type=("build", "run"))
+    depends_on("py-shellingham@1.1:1", when="@:1.1", type=("build", "run"))
+    depends_on("py-tomlkit@0.11.4:0", when="@1.6.1", type=("build", "run"))
+    depends_on("py-tomlkit@0.11.1,0.11.4:0", when="@1.2", type=("build", "run"))
     depends_on("py-tomlkit@0.7:0", when="@:1.1", type=("build", "run"))
-    depends_on("py-tomlkit@0.11.1,0.11.4:0", when="@1.2:", type=("build", "run"))
+    depends_on("py-virtualenv@20.22.0:20", when="@1.6.1", type=("build", "run"))
+    depends_on("py-virtualenv@20.4.3:20.4.4,20.4.7:", when="@1.2", type=("build", "run"))
     depends_on("py-virtualenv@20.0.26:20", when="@:1.1", type=("build", "run"))
-    depends_on("py-virtualenv@20.4.3:20.4.4,20.4.7:", when="@1.2:", type=("build", "run"))
+    depends_on("py-xattr@0.10", when="platform=darwin @1.6.1", type=("build", "run"))
     depends_on("py-xattr@0.9.7:0.9", when="platform=darwin @1.2:")
-    depends_on("py-urllib3@1.26.0:1", when="@1.2:")
+    depends_on("py-urllib3@1.26.0:1", when="@1.2")
+    depends_on("py-dulwich@0.21.2:0.21", when="@1.6.1", type=("build", "run"))
+    depends_on("py-dulwich@0.20.46:0.20", when="@1.2.1")
     depends_on("py-dulwich@0.20.44:0.20", when="@1.2.0")
-    depends_on("py-dulwich@0.20.46:0.20", when="@1.2.1:")
+    depends_on("py-build@0.10.0:0", when="@1.6.1", type=("build", "run"))
```

Comment:
```suggestion
    depends_on("py-build@0.10", when="@1.6.1", type=("build", "run"))
```
See https://python-poetry.org/docs/dependency-specification/#caret-requirements